include/ensmallen_bits/bigbatch_sgd/adaptive_stepsize.hpp

```diff
@@ -126,20 +126,24 @@ class AdaptiveStepsize
 
     // Update sample variance & norm of the gradient.
     sampleVariance = vB;
-    gradientNorm = std::pow(arma::norm(gradient / backtrackingBatchSize, 2), 2.0);
+    gradientNorm = std::pow(arma::norm(gradient / backtrackingBatchSize, 2),
+        2.0);
 
-    // Compute curvature.
-    double v = arma::trace(arma::trans(iterate - iteratePrev) *
-        (gradient - gradPrevIterate)) /
-        std::pow(arma::norm(iterate - iteratePrev, 2), 2.0);
+    // Compute curvature.  If it can't be computed (typically due to floating
+    // point representation issues), call it 0.  If the curvature is 0, the step
+    // size will not decay.
+    const double vNum = arma::trace(arma::trans(iterate - iteratePrev) *
+        (gradient - gradPrevIterate));
+    const double vDenom = std::pow(arma::norm(iterate - iteratePrev, 2), 2.0);
+    const double v = (vNum == 0.0 && vDenom == 0.0) ? 0.0 : (vNum / vDenom);
```

Comment:
From purely numerical perspective (not taking into account the nature of the algorithm), there's the possibility of `vNum / 0 = Inf`, so maybe this would be more robust:  
`const double v = (vDenom == 0.0) ? 0.0 : (vNum / vDenom);`

Alternatively, to take into account more possible corner cases, perhaps the check should be more general:  
`const double v_tmp = vNum / vDenom;`  
`const double v = std::is_finite(v_tmp) ? v_tmp : 0.0;`

Reply:
I think the only way we could have such a situation would be if floating-point errors caused `arma::norm(iterate - iteratePrev, 2)^2` to be 0, but `iterate - iteratePrev` was not 0.  In any case, yeah, probably good to handle that, and I think your solution that any non-finite vNum / vDenom should result in us taking `0.0` instead.